open-craft/opencraft
# [BB-3778] Implementing Static page for Dashboard

We're fulfilling the following user story:

> "As on OpenCraft Developer I want to implement static preview for dashboard page, so that user can get a real-time feel of how the customization will look like"

**JIRA tickets**:  [BB-3778](https://tasks.opencraft.com/browse/BB-3778), [Gitlab issue](https://gitlab.com/opencraft/dev/opencraft/-/issues/711)

**Dependencies**: None

**Screenshots**: See Gitlab issue

**Merge deadline**: "None"

**Testing instructions**:

1. Pull this branch to your local OCIM environment
2. Log with a test account; if you don't have one,  put a return on top of `validate_available_subdomain` in `registration/models.py`, and a dummy string in` lmsUrl = props.activeInstance.data.lmsUrl`; on `frontend/src/ui/components/Header/Header.tsx`. That should be enough for you to create an account. Then, make sure that account's appserver is successfully provisioned and there are both an instance reference and a Open edX deployment for your user. You can do this on /admin. Then, go to http://localhost:3000/console/theming/preview-and-colors
3. To test the dropdown, add:
```
  courseList: {
    defaultMessage: 'Course List',
    description: 'Course List dropdown item'
  }
```
to PreviewDropdown's `displayMessages.ts` so there's another option, which will hide the current preview component.

**Author notes and concerns**:

1. This should be merged before [BB-3787](https://tasks.opencraft.com/browse/BB-3787) is started or after it's completed (which would require a minor update) 
2. We're changing the ThemeAndColorPreviewComponent's name, but not its routing.
3. Currently we have a snapshot test; I'd implement a unit test for clicking the dropdown but that will be better implemented after [BB-3583](https://tasks.opencraft.com/browse/BB-3583) is merged. 

**Reviewers**
